### ISSUE TITLE
Editing Dim to Set does not give a compiler error

### ISSUE THREAD

**Issue by FullValueRider** [twinbasic/twinbasic#1487]
**Describe the bug**
in twinBasic we now have the facility to declare a variable and instantiate on a single line

```
Dim myC as Collection = new Collection
```

However, when refactoring it is possible to change from Dim to Set due to the variable being instantiated earlier in the Method
so in error we can end up with

```
Set myC as Collection = New Collection
```

The failure to delete the extraneous 'as' is not flagged as an error by the compiler and the line is compiled.  
At runtime, code using myC (as above) fails with a object not set error.  This has happened to me a few times over the past months and has led to a bit of headscratching until you read the code for what it is and not what you think it is.  Its even worse for primitive types where you don't get an object not set error..

**Expected behaviour**
The compiler should produce an error for the extraneous 'as' part.

**Desktop (please complete the following information):**
 - OS: WINDOWS 11
 - twinBASIC compiler version IDE Beta 246

**Comment by fafalone** [comment]
How are you changing it... if I select 'Dim' and overwrite it by typing 'Set', it recognizes the error and says ' Unrecognized symbol 'Collection'

**Comment by FullValueRider** [comment]
Good catch.  I simplified the issue by using a VBA native class.  As you report I get an error when I do this

```
Sub Test()

    Dim myC As Collection = New Collection
    Dim myc2 As Collection = New Collection
    myC.Add 10, "Hello"
    myC.Add 20, "There"
    myC.Add 30, "World"
    Set myC2 as collection = myC  ' collection is highlighted as an error
End Sub
```

If, however, I use a custom class with a PredeclaredId and a self factory method (in my case the self factory method is always 'Deb') then I don't see the error.
In the code below, KvpL is a Key Value Pair dictionary using Linked Lists as the host for the keys and values

```
Sub TestKvpL()

    Dim myK As KvpL = KvpL.Deb
    Dim myK2 As KvpL = KvpL.Deb
    myK.Add "Hello", 10
    myK.Add "There", 20
    myK.Add "World", 30
    Set myK2 as KvpL = myK
End Sub
```

![SetWithAs](https://user-images.githubusercontent.com/81257042/218414742-bc827bda-ebbe-488a-b081-8732aabee99b.jpg)

Most of the classes I use are custom classes so the only time I use new is in the construct instance method.
